### ISSUE TITLE
I'm contributing with a Spanish (es_419) translation for this project

### ISSUE THREAD

**Issue by tmarplatt** [vkbo/novelWriter#911]
Hello. I'm opening this ticket merely to inform that I am starting a full translation of novelWriter to the Spanish language (Latin American flavour). novelWriter is such an amazing piece of literary software, and I want to help its adoption among the LatAm community.

Since I do not know how much time it will take before submitting a PR, I'm opening this issue in case other contributors search the Issue tracker for the language name. You can find [my novelWriter fork here.](https://github.com/tmarplatt/novelWriter)

Any comments, advice, it's all welcome.

**Comment by vkbo** [comment]
Also, I just looked at your fork. Please don't commit directly to the main branch. That causes all sorts of issues when merging the changes. Please make a local branch instead. Please see the [Contributing Guide](https://github.com/vkbo/novelWriter/blob/main/CONTRIBUTING.md).

**Comment by tmarplatt** [comment]
Very well.

I noticed that QGroupBox title length actually pushes the boundaries of its parent box outwards, so e.g. a title long enough ([like the topmost title here](https://i.imgur.com/ardR3a2.png)) introduces a horizontal scrollbar. Yet such extended length by itself should not be enough to overflow the parent panel's width. Maybe it's a QT thing, maybe some additional padding is being calculated. I don't know if you can fix that.

**Comment by vkbo** [comment]
Yes, labels do not wrap by default, so if there is any way you can change that label to something shorter that still makes sense for formatting titles, please do so. This is one of the cases where I think the translation from English doesn't have to match exactly and instead should be a meaningful grouping label with a similar meaning.

On some locations I have hard coded an upper size limit where text will eventually be cut if the translations are too long. Otherwise it will mess up the GUI. I think it's mostly on tab labels, but keep an eye out for those too. I added them when the Portuguese translation was added. The latin-based languages can be a bit verbose at times compared to English and Norwegian.

As for the changes of PR #904, I don't think it should affect the translations. It's mostly a cleanup PR that changes which string formatting method is used as the default (since I no longer support Python 3.5) and improves test coverage.

**Comment by vkbo** [comment]
This was added by @bkmeneguello, so I never looked into the difference in detail, but I don't think the `lupdate` tool can be used directly. It doesn't parse all valid Python string formattings as far as I know.

**Comment by tmarplatt** [comment]
So it's been a while since I've written any Python code, but [I made this setup.py hack](https://github.com/tmarplatt/novelWriter/commit/364d6e2bc396a6135ef6da047a155aab78a4b3fe) that allows running qtlupdate as usual with the ability to retain translator comments: it stores them in a list of `xml.etree.ElementTree.Element` elements. And it only re-inserts comments belonging to strings that remain unchanged after running `pylupdate5`.

I tested it with the [practically finished es_419 file](https://github.com/tmarplatt/novelWriter/blob/es_419/i18n/nw_es_419.ts) and no errors, all comments remain, XML structure and format is retained. What do you think? @vkbo 

Finally, I have a list of string change suggestions coming up, perhaps those should go into a separate issue thread?

**Comment by vkbo** [comment]
It's decent Python code. Just a few redundancies related to scope of objects. I'm guessing you're more used to C/C++? Anyway, nothing that can't be easily fixed.

That said, I'm still uncertain about adding such a hack. Especially since it makes a few assumptions about the file content and structure without checking file version. I would probably rewrite it a little before taking it, but it may be an unnecessary duplication of effort. The issue is fixed in the [PyQt6's pylupdate](https://pypi.org/project/PyQt5/5.15.6/#files), but unfortunately not backported to PyQt5. The entire tool seems to have been rewritten. It may be simpler to just include the code from PyQt6 and put it in the `setup` directory. That way, we control how these files are generated and by what version. The pylupdate tool and novelWriter use the same license, so including it isn't an issue either.

Just make your PR with the TS files including comments, and I'll use that as a basis for testing that I can preserve them with the PyQt6 code. If we can't, then we'll merge your changes to the `setup.py` file and just use that hack instead.

**Comment by vkbo** [comment]
Ok, I've added the PyQt6 version of `pylupdate` in PR #920, but there are some minor issues. The main one being that the tool uses a different indentation of the TS XML than the Qt Linguist tool does. This will generate rather large diffs if files are added without first being cleaned up with Linguist. It may not be a big issue though. I may add a simple fix that reindents the XML after it's been generated.

I also noticed that the TS file format 2.1 will allow multiple `location` references in the XML. This means you assumption that `translatorcomment` should be inserted at index point 2 does not hold.
